### DISCUSSION TITLE
`pd not found` error while `check_estimator` used for estimator test

### DISCUSSION THREAD

**Discussion by afzal442** [alan-turing-institute/sktime#4660] [Q&A]
while I was working on `simple_linear_reg_forecaster`, I encountered an error `pd` not found; really hard to debug further.
here is my below code which I've been trying out

```py
import numpy as np
import pandas as pd
from sklearn.linear_model import LinearRegression
from sklearn.base import clone
from sktime.forecasting.base import BaseForecaster
from sktime.utils.validation.forecasting import check_fh

class SimpleReg(BaseForecaster):

    _tags = {
        "ignores-exogeneous-X": True,
        "requires-fh-in-fit": False,
        "handles-missing-data": False,
    }
        
    def __init__(self, regressor=None):
        """
        Initialize the SimpleReg forecaster.

        Parameters
        ----------
        regressor : object, optional (default=None)
            Custom regressor to use for linear regression.
            If None, LinearRegression(fit_intercept=True) will be used.
        """
        self.regressor = regressor
        super(SimpleReg, self).__init__()

    def _fit(self, y, X=None, fh=None):
        """
        Fit the forecaster to the training data using linear regression.

        Parameters
        ----------
        y : pd.Series
            Target time series with which to fit the forecaster.
        X : pd.DataFrame, default=None
            Exogenous variables are ignored.
        fh : int, list or np.array, optional (default=None)
            The forecast horizon with the steps ahead to predict.

        Returns
        -------
        self : reference to self
        """
        if self.regressor is None:
            self.regressor_ = LinearRegression(fit_intercept=True)
        else:
            self.regressor_ = clone(self.regressor)

        X_sklearn = _get_X_numpy_int_from_pandas(y.index)

        # Fit regressor
        self.regressor_.fit(X_sklearn, y)
        return self

    def _predict(self, fh=None, X=None):
        """
        Make forecasts for the given forecast horizon.

        Parameters
        ----------
        fh : int, list or np.array
            The forecast horizon with the steps ahead to predict.
        X : pd.DataFrame, default=None
            Exogenous variables (ignored).

        Returns
        -------
        y_pred : pd.Series
            Point predictions for the forecast.
        """
        if not hasattr(self, "regressor_"):
            raise NotFittedError("Model must be fitted before making predictions.")

        fh = check_fh(fh)

        X_sklearn = _get_X_numpy_int_from_pandas(fh)
        y_pred_sklearn = self.regressor_.predict(X_sklearn)
        y_pred = pd.Series(y_pred_sklearn, index=fh)
        y_pred.name = self._y.name
        return y_pred

def _get_X_numpy_int_from_pandas(index):
    """Convert pandas index to numpy array of integers"""
    return np.arange(len(index)).reshape(-1, 1)
```
anything I am missing here. LMK Thanks

**Comment by yarnabrina** [comment]
Is it a forecaster you're trying to add? Then I doubt `bug` is an appropriate label. May be we should move it from issue to discussions.

Can you share how are you trying to run `check_estimator`? You shared your codes here, but not where it exists nor how are you running the function that leads to the error. I'll suggest to provide the following information:

1. location of this code
2. exact code you are using to test this forecaster
3. the full error traceback that you received

Plus, as bug template suggests, versions of `python`, `sktime` and other packages can come in handy, but probably optional in this case.

**Reply by afzal442** [reply]
oh sorry for the confusion! that's any forecaster which I am trying to compile `Linear Reg model` to be fit using our extension template.
I ran sktime `check_estimator` utility to test that abv estimator.
```py
from sktime.utils.estimator_checks import check_estimator
from sktime.forecasting.simple_linear_reg_forecaster import SimpleReg
check_estimator(SimpleReg)
```
Hope that makes sense now. Thanks

**Reply by yarnabrina** [reply]
Can you provide location and traceback, as per points 1 and 3 above? It'll be better if you eidt the original question with full details instead of in comments.

Please take a look at [MRE](https://stackoverflow.com/help/minimal-reproducible-example) for future questions.

**Reply by yarnabrina** [reply]
@afzal442 Based on your comments about code location in discord, I did create a module `sktime/forecasting/simple_linear_reg_forecaster.py` and added above code there. Running the tests led to a lot of failures (106 passed, and 156 failed). All the failures complained about `AssertionError`, and I did not get any error related to `pd`.

Few things that you should check: are you in the right environment? does your environment have `pandas`? what does the error traceback show? Can you spot the line where the error originated?

Also, the estimator you are trying to create is covered by `TrendForecaster` I believe. Please check that one out, and confirm if what you want to add is not a duplicate.

**Comment by fkiraly** [comment]
agree, it is also labelled discussion by @afzal442 - so, let's move to discussions for now.
